### DISCUSSION TITLE
customManagers with enabledManagers

### DISCUSSION THREAD

**Discussion by S73FAIV** [renovatebot/renovate#25898] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.64.2

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

I have a special use case, where I want to use only a custom regex manager with renovate.

For this I defined my custom manager as documented:
```
"customManagers": [
    {
      "customType": "regex",
      "fileMatch": [
        ".*\\/some-file-names\\.ya?ml$"
      ],
      "matchStrings": [
        "^# renovate: datasource=(?<datasource>[a-z-.]+?) depName=(?<depName>[^\\s]+?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[^\\s]+?))?(?: registryUrl=(?<registryUrl>[^\\s]+?))?\\s*.*\\s*(version|tag): \\s*?(?<currentValue>.+?)?$"
      ]
    }
```

and used the `"enabledManagers": ["custom.regex"],` to let renovate only use the created custom manager.

But this leads to some irritating behavior, as seen in the logs.
The _fileMatch_ works just fine and finds the files I tell him to, but afterwords, he doesn't extract the dependencies.
I checked the _matchStrings_ regex multiple times and there doesn't seem to be an error either.

What would be a viable configuration to achieve the same result?

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: Using file match: .*\/\.ya?ml$ for manager regex
DEBUG: Matched 12 file(s) for manager regex: ..omitted...
DEBUG: manager extract durations (ms) 
       "managers": {"regex": 3}
DEBUG: Found 0 package file(s)
DEBUG: Manager explicitly enabled in "enabledManagers" config, but found no results. Possible config error? 
       "manager": "custom.regex"
 INFO: Dependency extraction complete 
       "stats": {"managers": {}, "total": {"fileCount": 0, "depCount": 0}}

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by S73FAIV** [comment]
Does this maybe have a connection to the still open points in this issue: #19066 ?

**Reply by rarkins** [reply]
Seems to be, please try "custom" or "regex" for now

**Comment by S73FAIV** [comment]
I provided all enbaledManagers as CLI-Options, because otherwise _regex_ would be migrated to _custom.regex_

So with *renovate --dry-run --enabled-managers "regex","custom","custom.regex"*

I get the following config: ` "enabledManagers": ["regex", "custom", "custom.regex"],`

With the same behavior as before:
```
DEBUG: Using file match: .*\/.*\.ya?ml$ for manager regex
DEBUG: Matched 12 file(s) for manager regex: **omitted**
DEBUG: manager extract durations (ms)
       "managers": {"regex": 4}
DEBUG: Found 0 package file(s) 
DEBUG: Manager explicitly enabled in "enabledManagers" config, but found no results. Possible config error? 
       "manager": "regex"
DEBUG: Manager explicitly enabled in "enabledManagers" config, but found no results. Possible config error? 
       "manager": "custom"
DEBUG: Manager explicitly enabled in "enabledManagers" config, but found no results. Possible config error? 
       "manager": "custom.regex"
 INFO: Dependency extraction complete
```

**Comment by S73FAIV** [comment]
Ah, I have to apologize! The error was on my side, my _matchString_ regex was faulty, which lead to the error encountered.
The enabledManagers works fine with customManager in the first suggested Version
`"enabledManagers": ["custom.regex"]`

Thank you very much for your support nevertheless.
